In [43]:
import pandas as pd
import plotly.express as px

In [44]:
# Read the dataset 'quakes-cleaned.csv' into a pandas DataFrame.
# Use the values in the first column as the index for the DataFrame.
df = pd.read_csv('quakes-cleaned - Copy.csv', index_col=0)

In [45]:
df.head()

latitude   longitude      depth   mag  \
time                                                                       
2023-11-15 09:05:07.304000+00:00  61.580800 -149.847000  32.800000  1.70   
2023-11-15 08:53:06.688000+00:00  61.079400 -147.883000  14.800000  1.00   
2023-11-15 08:41:52.480000+00:00  19.380667 -155.285339   0.320000  1.73   
2023-11-15 07:44:53.035000+00:00  61.638200 -149.782800  32.900000  1.90   
2023-11-15 07:19:44.540000+00:00  18.972166 -155.451660  34.759998  1.87   

                                 magType  nst    gap  dmin   rms net  ...  \
time                                                                  ...   
2023-11-15 09:05:07.304000+00:00      ml    0    0.0   0.0  0.22  ak  ...   
2023-11-15 08:53:06.688000+00:00      ml    0    0.0   0.0  0.80  ak  ...   
2023-11-15 08:41:52.480000+00:00      md   15  153.0   0.0  0.20  hv  ...   
2023-11-15 07:44:53.035000+00:00      ml    0    0.0   0.0  0.31  ak  ...   
2023-11-15 07:19:44.540000+00:00      md   37  236.0   0.0  0.12  hv  ...   

                                                         place        type  \
time                                                                         
2023-11-15 09:05:07.304000+00:00   5 km SSW of Houston, Alaska  earthquake   
2023-11-15 08:53:06.688000+00:00  55 km NE of Whittier, Alaska  earthquake   
2023-11-15 08:41:52.480000+00:00    8 km SW of Volcano, Hawaii  earthquake   
2023-11-15 07:44:53.035000+00:00                        Others  earthquake   
2023-11-15 07:19:44.540000+00:00   17 km SE of Naalehu, Hawaii  earthquake   

                                 horizontalError depthError  magError  magNst  \
time                                                                            
2023-11-15 09:05:07.304000+00:00            0.00       0.20      0.00       0   
2023-11-15 08:53:06.688000+00:00            0.00       0.30      0.00       0   
2023-11-15 08:41:52.480000+00:00            0.33       0.38      0.59      15   
2023-11-15 07:44:53.035000+00:00            0.00       0.20      0.00       0   
2023-11-15 07:19:44.540000+00:00            0.71       0.89      0.88       5   

                                     status  locationSource magSource  \
time                                                                    
2023-11-15 09:05:07.304000+00:00  automatic              ak        ak   
2023-11-15 08:53:06.688000+00:00  automatic              ak        ak   
2023-11-15 08:41:52.480000+00:00  automatic              hv        hv   
2023-11-15 07:44:53.035000+00:00  automatic              ak        ak   
2023-11-15 07:19:44.540000+00:00  automatic              hv        hv   

                                                                      state_country  
time                                                                                 
2023-11-15 09:05:07.304000+00:00  Beavertail Drive, Matanuska-Susitna, Alaska, U...  
2023-11-15 08:53:06.688000+00:00                     Chugach, Alaska, United States  
2023-11-15 08:41:52.480000+00:00  Volcano, Volcano CDP, Hawaiʻi County, Hawaii, ...  
2023-11-15 07:44:53.035000+00:00  Houston, Matanuska-Susitna, Alaska, 99694, Uni...  
2023-11-15 07:19:44.540000+00:00                                      United States  

[5 rows x 22 columns]

# 1

In [46]:
df.type.value_counts()

earthquake                18581
quarry blast                270
ice quake                   230
explosion                   134
other event                  17
volcanic eruption             9
experimental explosion        1
landslide                     1
mining explosion              1
Name: type, dtype: int64

**During the two-and-a-half-month period, seismic events were predominantly earthquakes, accounting for approximately 96% of the total occurrences.**

**Specifically, earthquake events occurred 18,581 times, representing the majority of recorded incidents, while other types, such as volcanic eruptions, experimental explosions, landslides, and mining explosions, were relatively rare, each occurring only a handful of times.**

In [47]:
# Get the value counts for the 'type' column
type_counts = df['type'].value_counts().reset_index()
type_counts.columns = ['type', 'count']

# Plotting the bar chart using Plotly Express
fig = px.bar(type_counts, x='type', y='count', title='Seismic Event Distribution',
             labels={'type': 'Event Type', 'count': 'Event Frequency'})

# Show the plot
fig.show()

# 2. 

In [48]:
df

latitude   longitude      depth   mag  \
time                                                                       
2023-11-15 09:05:07.304000+00:00  61.580800 -149.847000  32.800000  1.70   
2023-11-15 08:53:06.688000+00:00  61.079400 -147.883000  14.800000  1.00   
2023-11-15 08:41:52.480000+00:00  19.380667 -155.285339   0.320000  1.73   
2023-11-15 07:44:53.035000+00:00  61.638200 -149.782800  32.900000  1.90   
2023-11-15 07:19:44.540000+00:00  18.972166 -155.451660  34.759998  1.87   
...                                     ...         ...        ...   ...   
2023-09-01 01:10:27.784000+00:00  53.154800 -164.532800  35.000000  3.00   
2023-09-01 00:55:24.900000+00:00  19.210167 -155.348007  28.299999  1.71   
2023-09-01 00:45:43.100000+00:00  17.937167  -66.917667  13.160000  2.28   
2023-09-01 00:42:08.428000+00:00  60.279600 -147.859600   5.600000  2.00   
2023-09-01 00:04:57.310000+00:00  19.132000 -155.375336  29.879999  2.01   

                                 magType  nst    gap    dmin   rms net  ...  \
time                                                                    ...   
2023-11-15 09:05:07.304000+00:00      ml    0    0.0  0.0000  0.22  ak  ...   
2023-11-15 08:53:06.688000+00:00      ml    0    0.0  0.0000  0.80  ak  ...   
2023-11-15 08:41:52.480000+00:00      md   15  153.0  0.0000  0.20  hv  ...   
2023-11-15 07:44:53.035000+00:00      ml    0    0.0  0.0000  0.31  ak  ...   
2023-11-15 07:19:44.540000+00:00      md   37  236.0  0.0000  0.12  hv  ...   
...                                  ...  ...    ...     ...   ...  ..  ...   
2023-09-01 01:10:27.784000+00:00      ml   14  222.0  1.2290  0.33  us  ...   
2023-09-01 00:55:24.900000+00:00      md   21  189.0  0.0000  0.10  hv  ...   
2023-09-01 00:45:43.100000+00:00      md    6  216.0  0.0525  0.08  pr  ...   
2023-09-01 00:42:08.428000+00:00      ml    0    0.0  0.0000  0.74  ak  ...   
2023-09-01 00:04:57.310000+00:00      md   36  192.0  0.0000  0.09  hv  ...   

                                                         place        type  \
time                                                                         
2023-11-15 09:05:07.304000+00:00   5 km SSW of Houston, Alaska  earthquake   
2023-11-15 08:53:06.688000+00:00  55 km NE of Whittier, Alaska  earthquake   
2023-11-15 08:41:52.480000+00:00    8 km SW of Volcano, Hawaii  earthquake   
2023-11-15 07:44:53.035000+00:00                        Others  earthquake   
2023-11-15 07:19:44.540000+00:00   17 km SE of Naalehu, Hawaii  earthquake   
...                                                        ...         ...   
2023-09-01 01:10:27.784000+00:00   136 km SE of Akutan, Alaska  earthquake   
2023-09-01 00:55:24.900000+00:00     13 km E of Pāhala, Hawaii  earthquake   
2023-09-01 00:45:43.100000+00:00            Puerto Rico region  earthquake   
2023-09-01 00:42:08.428000+00:00  25 km NNE of Chenega, Alaska  earthquake   
2023-09-01 00:04:57.310000+00:00    13 km SE of Pāhala, Hawaii  earthquake   

                                 horizontalError depthError  magError  magNst  \
time                                                                            
2023-11-15 09:05:07.304000+00:00            0.00       0.20  0.000000       0   
2023-11-15 08:53:06.688000+00:00            0.00       0.30  0.000000       0   
2023-11-15 08:41:52.480000+00:00            0.33       0.38  0.590000      15   
2023-11-15 07:44:53.035000+00:00            0.00       0.20  0.000000       0   
2023-11-15 07:19:44.540000+00:00            0.71       0.89  0.880000       5   
...                                          ...        ...       ...     ...   
2023-09-01 01:10:27.784000+00:00            3.27       2.02  0.128000       8   
2023-09-01 00:55:24.900000+00:00            0.82       0.63  1.730000       4   
2023-09-01 00:45:43.100000+00:00            0.90       0.41  0.130183       6   
2023-09-01 00:42:08.428000+00:00            0.00       0.40  0.000000       0   
2023-09-01 00:04:57.310000+00:0

In [49]:
df.mag.describe()

count    19244.000000
mean         2.156412
std          1.141293
min          1.000000
25%          1.320000
50%          1.800000
75%          2.470000
max          7.100000
Name: mag, dtype: float64

Between September 1st, 2023, and 9:30 am on November 15th, 2023, 19,244 earthquakes were recorded. On average, earthquakes had a magnitude of approximately 2.156, indicating a moderate strength. The variability around this average is captured by a standard deviation of 1.141. Quartiles reveal that the central 50% of earthquakes fell within the range of 1.320 to 2.470. The dataset shows a diverse range of earthquake magnitudes, with a minimum of 1.000 and a maximum of 7.100.

In [50]:
# Plotting the box plot for the 'mag' column
fig = px.box(df, y='mag', title='Box Plot of Magnitude',
             labels={'mag': 'Magnitude'})

# Show the plot
fig.show()

The earthquakes recorded between September 1st, 2023, and 9:30 am on November 15th, 2023, exhibit a diverse range of magnitudes. On average, earthquakes during this period had a magnitude of approximately 2.16. The interquartile range (IQR), representing the middle 50% of the data, spans from 1.32 to 2.47. The median magnitude (50th percentile) is 1.80, while the 25th and 75th percentiles are 1.32 and 2.47, respectively. The standard deviation is about 1.14, indicating moderate variability. Outliers, represented as individual points beyond the whiskers in the box plot, suggest instances of more intense seismic events, with the maximum recorded magnitude reaching 7.10. Monitoring and understanding these magnitudes are crucial for assessing seismic risks and implementing appropriate safety measures.

# 3. 

In [51]:
df.head()

latitude   longitude      depth   mag  \
time                                                                       
2023-11-15 09:05:07.304000+00:00  61.580800 -149.847000  32.800000  1.70   
2023-11-15 08:53:06.688000+00:00  61.079400 -147.883000  14.800000  1.00   
2023-11-15 08:41:52.480000+00:00  19.380667 -155.285339   0.320000  1.73   
2023-11-15 07:44:53.035000+00:00  61.638200 -149.782800  32.900000  1.90   
2023-11-15 07:19:44.540000+00:00  18.972166 -155.451660  34.759998  1.87   

                                 magType  nst    gap  dmin   rms net  ...  \
time                                                                  ...   
2023-11-15 09:05:07.304000+00:00      ml    0    0.0   0.0  0.22  ak  ...   
2023-11-15 08:53:06.688000+00:00      ml    0    0.0   0.0  0.80  ak  ...   
2023-11-15 08:41:52.480000+00:00      md   15  153.0   0.0  0.20  hv  ...   
2023-11-15 07:44:53.035000+00:00      ml    0    0.0   0.0  0.31  ak  ...   
2023-11-15 07:19:44.540000+00:00      md   37  236.0   0.0  0.12  hv  ...   

                                                         place        type  \
time                                                                         
2023-11-15 09:05:07.304000+00:00   5 km SSW of Houston, Alaska  earthquake   
2023-11-15 08:53:06.688000+00:00  55 km NE of Whittier, Alaska  earthquake   
2023-11-15 08:41:52.480000+00:00    8 km SW of Volcano, Hawaii  earthquake   
2023-11-15 07:44:53.035000+00:00                        Others  earthquake   
2023-11-15 07:19:44.540000+00:00   17 km SE of Naalehu, Hawaii  earthquake   

                                 horizontalError depthError  magError  magNst  \
time                                                                            
2023-11-15 09:05:07.304000+00:00            0.00       0.20      0.00       0   
2023-11-15 08:53:06.688000+00:00            0.00       0.30      0.00       0   
2023-11-15 08:41:52.480000+00:00            0.33       0.38      0.59      15   
2023-11-15 07:44:53.035000+00:00            0.00       0.20      0.00       0   
2023-11-15 07:19:44.540000+00:00            0.71       0.89      0.88       5   

                                     status  locationSource magSource  \
time                                                                    
2023-11-15 09:05:07.304000+00:00  automatic              ak        ak   
2023-11-15 08:53:06.688000+00:00  automatic              ak        ak   
2023-11-15 08:41:52.480000+00:00  automatic              hv        hv   
2023-11-15 07:44:53.035000+00:00  automatic              ak        ak   
2023-11-15 07:19:44.540000+00:00  automatic              hv        hv   

                                                                      state_country  
time                                                                                 
2023-11-15 09:05:07.304000+00:00  Beavertail Drive, Matanuska-Susitna, Alaska, U...  
2023-11-15 08:53:06.688000+00:00                     Chugach, Alaska, United States  
2023-11-15 08:41:52.480000+00:00  Volcano, Volcano CDP, Hawaiʻi County, Hawaii, ...  
2023-11-15 07:44:53.035000+00:00  Houston, Matanuska-Susitna, Alaska, 99694, Uni...  
2023-11-15 07:19:44.540000+00:00                                      United States  

[5 rows x 22 columns]

In [52]:
df.status.value_counts()

reviewed     16801
automatic     2443
Name: status, dtype: int64

In [53]:
# Plotting the pie chart for the 'status' column
fig = px.pie(df, names='status', title='Distribution of Earthquake Status',
             labels={'status': 'Status'}, color_discrete_sequence=px.colors.sequential.RdBu)

# Show the plot
fig.show()

The pie chart reveals that the majority (87.3%) of seismic events in the dataset have undergone human review, indicating a meticulous curation process for data accuracy. The remaining 12.7% are labeled as "automatic," suggesting the utilization of automated systems for swift identification and recording of seismic events. This distribution underscores the combined use of automated processes and human expertise in maintaining data quality and reliability.

## 4.

In [25]:
df.head()

latitude   longitude      depth   mag  \
time                                                                       
2023-11-15 09:05:07.304000+00:00  61.580800 -149.847000  32.800000  1.70   
2023-11-15 08:53:06.688000+00:00  61.079400 -147.883000  14.800000  1.00   
2023-11-15 08:41:52.480000+00:00  19.380667 -155.285339   0.320000  1.73   
2023-11-15 07:44:53.035000+00:00  61.638200 -149.782800  32.900000  1.90   
2023-11-15 07:19:44.540000+00:00  18.972166 -155.451660  34.759998  1.87   

                                 magType  nst    gap  dmin   rms net  ...  \
time                                                                  ...   
2023-11-15 09:05:07.304000+00:00      ml    0    0.0   0.0  0.22  ak  ...   
2023-11-15 08:53:06.688000+00:00      ml    0    0.0   0.0  0.80  ak  ...   
2023-11-15 08:41:52.480000+00:00      md   15  153.0   0.0  0.20  hv  ...   
2023-11-15 07:44:53.035000+00:00      ml    0    0.0   0.0  0.31  ak  ...   
2023-11-15 07:19:44.540000+00:00      md   37  236.0   0.0  0.12  hv  ...   

                                                         place        type  \
time                                                                         
2023-11-15 09:05:07.304000+00:00   5 km SSW of Houston, Alaska  earthquake   
2023-11-15 08:53:06.688000+00:00  55 km NE of Whittier, Alaska  earthquake   
2023-11-15 08:41:52.480000+00:00    8 km SW of Volcano, Hawaii  earthquake   
2023-11-15 07:44:53.035000+00:00                        Others  earthquake   
2023-11-15 07:19:44.540000+00:00   17 km SE of Naalehu, Hawaii  earthquake   

                                 horizontalError depthError  magError  magNst  \
time                                                                            
2023-11-15 09:05:07.304000+00:00            0.00       0.20      0.00       0   
2023-11-15 08:53:06.688000+00:00            0.00       0.30      0.00       0   
2023-11-15 08:41:52.480000+00:00            0.33       0.38      0.59      15   
2023-11-15 07:44:53.035000+00:00            0.00       0.20      0.00       0   
2023-11-15 07:19:44.540000+00:00            0.71       0.89      0.88       5   

                                     status  locationSource magSource  \
time                                                                    
2023-11-15 09:05:07.304000+00:00  automatic              ak        ak   
2023-11-15 08:53:06.688000+00:00  automatic              ak        ak   
2023-11-15 08:41:52.480000+00:00  automatic              hv        hv   
2023-11-15 07:44:53.035000+00:00  automatic              ak        ak   
2023-11-15 07:19:44.540000+00:00  automatic              hv        hv   

                                                                      state_country  
time                                                                                 
2023-11-15 09:05:07.304000+00:00  Beavertail Drive, Matanuska-Susitna, Alaska, U...  
2023-11-15 08:53:06.688000+00:00                     Chugach, Alaska, United States  
2023-11-15 08:41:52.480000+00:00  Volcano, Volcano CDP, Hawaiʻi County, Hawaii, ...  
2023-11-15 07:44:53.035000+00:00  Houston, Matanuska-Susitna, Alaska, 99694, Uni...  
2023-11-15 07:19:44.540000+00:00                                      United States  

[5 rows x 22 columns]

In [26]:
df.magType.value_counts()

ml            11681
md             3815
mb             2453
ml(texnet)      872
mww             262
mwr              86
mlv              28
mb_lg            21
mw               19
mh                3
mwb               1
Mi                1
Mb                1
ms_vx             1
Name: magType, dtype: int64

In [27]:
# Plotting the scatter plot with a regression line using Plotly Express
fig = px.scatter(df, x='depth', y='dmin', title='Relationship between Depth and Magnitude with Regression Line',
                 labels={'depth': 'Depth', 'mag': 'Magnitude'},
                 trendline='ols')  # 'ols' stands for Ordinary Least Squares (linear regression)

# Show the plot
fig.show()

i have the following numneric columns in the dataset: depth, mag, nst, dmin, rms, horizontalError, depthError, magError, magNst

In [28]:
import plotly.graph_objects as go
# Calculate the correlation matrix
correlation_matrix = df.corr()

# Plotting the correlation matrix as a heatmap
fig = go.Figure(data=go.Heatmap(z=correlation_matrix.values,
                                x=correlation_matrix.columns,
                                y=correlation_matrix.columns,
                                colorscale='Viridis',
                                colorbar=dict(title='Correlation')))

# Show the plot
fig.update_layout(title='Correlation Matrix')
fig.show()

# 5.

In [29]:
df.head()

latitude   longitude      depth   mag  \
time                                                                       
2023-11-15 09:05:07.304000+00:00  61.580800 -149.847000  32.800000  1.70   
2023-11-15 08:53:06.688000+00:00  61.079400 -147.883000  14.800000  1.00   
2023-11-15 08:41:52.480000+00:00  19.380667 -155.285339   0.320000  1.73   
2023-11-15 07:44:53.035000+00:00  61.638200 -149.782800  32.900000  1.90   
2023-11-15 07:19:44.540000+00:00  18.972166 -155.451660  34.759998  1.87   

                                 magType  nst    gap  dmin   rms net  ...  \
time                                                                  ...   
2023-11-15 09:05:07.304000+00:00      ml    0    0.0   0.0  0.22  ak  ...   
2023-11-15 08:53:06.688000+00:00      ml    0    0.0   0.0  0.80  ak  ...   
2023-11-15 08:41:52.480000+00:00      md   15  153.0   0.0  0.20  hv  ...   
2023-11-15 07:44:53.035000+00:00      ml    0    0.0   0.0  0.31  ak  ...   
2023-11-15 07:19:44.540000+00:00      md   37  236.0   0.0  0.12  hv  ...   

                                                         place        type  \
time                                                                         
2023-11-15 09:05:07.304000+00:00   5 km SSW of Houston, Alaska  earthquake   
2023-11-15 08:53:06.688000+00:00  55 km NE of Whittier, Alaska  earthquake   
2023-11-15 08:41:52.480000+00:00    8 km SW of Volcano, Hawaii  earthquake   
2023-11-15 07:44:53.035000+00:00                        Others  earthquake   
2023-11-15 07:19:44.540000+00:00   17 km SE of Naalehu, Hawaii  earthquake   

                                 horizontalError depthError  magError  magNst  \
time                                                                            
2023-11-15 09:05:07.304000+00:00            0.00       0.20      0.00       0   
2023-11-15 08:53:06.688000+00:00            0.00       0.30      0.00       0   
2023-11-15 08:41:52.480000+00:00            0.33       0.38      0.59      15   
2023-11-15 07:44:53.035000+00:00            0.00       0.20      0.00       0   
2023-11-15 07:19:44.540000+00:00            0.71       0.89      0.88       5   

                                     status  locationSource magSource  \
time                                                                    
2023-11-15 09:05:07.304000+00:00  automatic              ak        ak   
2023-11-15 08:53:06.688000+00:00  automatic              ak        ak   
2023-11-15 08:41:52.480000+00:00  automatic              hv        hv   
2023-11-15 07:44:53.035000+00:00  automatic              ak        ak   
2023-11-15 07:19:44.540000+00:00  automatic              hv        hv   

                                                                      state_country  
time                                                                                 
2023-11-15 09:05:07.304000+00:00  Beavertail Drive, Matanuska-Susitna, Alaska, U...  
2023-11-15 08:53:06.688000+00:00                     Chugach, Alaska, United States  
2023-11-15 08:41:52.480000+00:00  Volcano, Volcano CDP, Hawaiʻi County, Hawaii, ...  
2023-11-15 07:44:53.035000+00:00  Houston, Matanuska-Susitna, Alaska, 99694, Uni...  
2023-11-15 07:19:44.540000+00:00                                      United States  

[5 rows x 22 columns]

In [30]:
df.place.value_counts()

Others                               972
Southern Alaska                      208
Island of Hawaii, Hawaii             175
Izu Islands, Japan region            173
Central Alaska                       145
                                    ... 
80 km NNW of Chalkyitsik, Alaska       1
29 km SE of Susitna North, Alaska      1
3 km WSW of Newbern, Tennessee         1
2 km ENE of Monterey Park, CA          1
25 km NNE of Chenega, Alaska           1
Name: place, Length: 10315, dtype: int64

In [31]:
df.shape

(19244, 22)

In [32]:
# Extract the text after the comma or the whole 'state' column if there is no comma
df['state'] = df['place'].apply(lambda x: x.split(', ')[-1] if isinstance(x, str) and ', ' in x else x)


In [33]:
df.state.value_counts()

Alaska                            6727
CA                                3399
Hawaii                            1321
Others                             972
Texas                              568
                                  ... 
Australia                            1
northern Colombia                    1
south of Australia                   1
Colombia-Ecuador border region       1
Martinique                           1
Name: state, Length: 259, dtype: int64

In [34]:
list(df.state.value_counts().index)

['Alaska',
 'CA',
 'Hawaii',
 'Others',
 'Texas',
 'Puerto Rico',
 'Oklahoma',
 'Nevada',
 'New Mexico',
 'Indonesia',
 'Southern Alaska',
 'Washington',
 'Montana',
 'Japan region',
 'Papua New Guinea',
 'Idaho',
 'Japan',
 'Central Alaska',
 'Utah',
 'western Texas',
 'Puerto Rico region',
 'Chile',
 'California',
 'Colorado',
 'Canada',
 'Philippines',
 'MX',
 'Tonga',
 'Russia',
 'Afghanistan',
 'northern Alaska',
 'Alaska Peninsula',
 'Argentina',
 'Fiji region',
 'Oregon',
 'Banda Sea',
 'Mexico',
 'U.S. Virgin Islands',
 'Wyoming',
 'China',
 'south of the Fiji Islands',
 'Solomon Islands',
 'South Sandwich Islands region',
 'Vanuatu',
 'New Zealand',
 'Tennessee',
 'Dominican Republic',
 'Iran',
 'Fiji',
 'Tajikistan',
 'Peru',
 'Timor Leste',
 'Northern Mariana Islands',
 'Missouri',
 'Kermadec Islands region',
 'Southeastern Alaska',
 'southern Idaho',
 'California-Nevada border region',
 'Greece',
 'Iceland',
 'Guam',
 'India',
 'south of the Kermadec Islands',
 'Arkansas',


In [35]:
a = df[['longitude', 'latitude', 'state', 'place']]

In [36]:
filtered_df = a[a['state'] == a['place']]
filtered_df.head(5)

longitude   latitude           state  \
time                                                                      
2023-11-15 07:44:53.035000+00:00 -149.782800  61.638200          Others   
2023-11-15 04:03:09.225000+00:00 -150.993800  62.401700  Central Alaska   
2023-11-15 03:32:35.251000+00:00 -147.792800  61.567000          Others   
2023-11-15 03:00:42.826000+00:00 -149.014700  62.353400          Others   
2023-11-15 02:42:01.054000+00:00  -97.324768  35.928886          Others   

                                           place  
time                                              
2023-11-15 07:44:53.035000+00:00          Others  
2023-11-15 04:03:09.225000+00:00  Central Alaska  
2023-11-15 03:32:35.251000+00:00          Others  
2023-11-15 03:00:42.826000+00:00          Others  
2023-11-15 02:42:01.054000+00:00          Others

In [37]:
# Grouping by 'place' and counting occurrences
df1 = df.groupby(['place']).size().reset_index(name='count_')

# If you want to keep only 'place', 'latitude', 'longitude', and 'count' columns
df1 = df1[['place', 'count_']]

# Display the resulting DataFrame
print(df1)

                                 place  count_
0      0 km E of Coleville, California       1
1      0 km E of Howardville, Missouri       1
2        0 km E of Indios, Puerto Rico       1
3       0 km E of Parkwood, Washington       1
4            0 km E of The Geysers, CA       1
...                                ...     ...
10310              western Afghanistan       8
10311            western Honshu, Japan       1
10312   western Indian-Antarctic Ridge       4
10313                    western Texas     130
10314                   western Xizang       1

[10315 rows x 2 columns]


In [38]:
df1.shape

(10315, 2)

In [39]:
# Merge the two DataFrames based on the 'place' column
result_df = pd.merge(df1, df[['place', 'latitude', 'longitude']], on='place', how='inner')
result_df.shape
# Display the resulting DataFrame


(19244, 4)

In [40]:
unique_places = df[['place', 'longitude', 'latitude']].drop_duplicates()

# Display the resulting DataFrame
unique_places.sample(10)


place  \
time                                                                  
2023-10-25 09:50:36.580000+00:00       6 km NW of Lepanto, Arkansas   
2023-10-10 10:56:14.816000+00:00         91 km WNW of Puerto, Chile   
2023-10-07 04:08:14.663000+00:00          54 km E of Juneau, Alaska   
2023-09-04 12:17:38.797000+00:00          4 km SW of Tyonek, Alaska   
2023-09-22 23:43:05.256000+00:00            71 km E of Atka, Alaska   
2023-11-11 17:04:15.390000+00:00          44 km WNW of Petrolia, CA   
2023-10-03 18:05:09.571000+00:00          66 km W of Tyonek, Alaska   
2023-10-07 03:07:16.320000+00:00  4 km WSW of San Juan Bautista, CA   
2023-09-21 00:00:35.260000+00:00        35 km S of Nelchina, Alaska   
2023-09-24 21:20:56.950000+00:00    60 km NNE of Chickaloon, Alaska   

                                   longitude   latitude  
time                                                     
2023-10-25 09:50:36.580000+00:00  -90.387333  35.648167  
2023-10-10 10:56:14.816000+00:00  -70.997900 -19.993500  
2023-10-07 04:08:14.663000+00:00 -133.490300  58.355000  
2023-09-04 12:17:38.797000+00:00 -151.201500  61.037900  
2023-09-22 23:43:05.256000+00:00 -173.173200  52.300100  
2023-11-11 17:04:15.390000+00:00 -124.793500  40.414167  
2023-10-03 18:05:09.571000+00:00 -152.358700  60.998000  
2023-10-07 03:07:16.320000+00:00 -121.582167  36.829500  
2023-09-21 00:00:35.260000+00:00 -146.643800  61.680600  
2023-09-24 21:20:56.950000+00:00 -148.158300  62.317600

In [ ]:
fig = px.scatter_geo(
    scope="world",  # The in-built plotly map to use.  There isn't one for Ireland.  :-(
    data_frame=df,  # The dataframe to source the data from.
    lon=df.longitude,  # The longitude points to plot (think of these as along the x-axis).
    lat=df.latitude,  # The latitude points to plot (think of these as along the y-axis).
    #size=df.Rain,  # Used to determine how big of a circle to draw at each point (where x meets y).
    hover_data=df,  # The data to display when an (x, y) point is hovered over.
    color=df.place,  # The colours to use for each of the points (with a different colour per Station).
    height=700,  # How big (height-wise) the graph should be when drawn.
)

# fig.update_geos(
#     resolution=50,  # Controls how detailed the drawn map is. The other choice is 110 (looks awful).
#     lonaxis={
#         "range": [-12, -5]
#     },  # The longitute range from the "europe" map to display.  Left and Right sides.
#     lataxis={
#         "range": [52, 55]
#     },  # The latitude range from the "europe" map to display.  Bottom and Top sides.
# )

fig.show()  # Displays the interactive plotly graphic on screen.+

In [ ]:
df1 = pd.read_csv('quakes-cleaned3.csv', index_col=0)

In [ ]:
df1.head()

In [ ]:
list(df1.state_country.head())

In [ ]:
# If you want to overwrite the existing column:
df1['state_country'] = df1['state_country'].apply(lambda x: x.split(',')[-1].strip() if ',' in x else x.strip())



In [ ]:
list(df1.state_country.head())

In [ ]:
df1.state_country.value_counts()

In [ ]:
#df1 = df1.sample(50)

In [ ]:
fig = px.scatter_geo(
    scope="world",  # The in-built plotly map to use.  There isn't one for Ireland.  :-(
    data_frame=df1,  # The dataframe to source the data from.
    lon=df1.longitude,  # The longitude points to plot (think of these as along the x-axis).
    lat=df1.latitude,  # The latitude points to plot (think of these as along the y-axis).
    #size=df.Rain,  # Used to determine how big of a circle to draw at each point (where x meets y).
    hover_data=df1,  # The data to display when an (x, y) point is hovered over.
    color=df1.state_country,  # The colours to use for each of the points (with a different colour per Station).
    #height=700,  # How big (height-wise) the graph should be when drawn.
    basemap_visible = True
 )

fig.update_geos(
    resolution=50,  # Controls how detailed the drawn map is. The other choice is 110 (looks awful).
    lonaxis={
        "range": [-180, 180]
    },  # The longitute range from the "europe" map to display.  Left and Right sides.
    lataxis={
        "range": [676, -4]
    },  # The latitude range from the "europe" map to display.  Bottom and Top sides.
)

fig.show()  # Displays the interactive plotly graphic on screen.+

In [ ]:
fig = px.scatter_geo(
    scope="world",
    data_frame=df1,
    lon=df1.longitude,
    lat=df1.latitude,
    hover_data=df1,
    color=df1.state_country,
    basemap_visible=True
)

fig.update_geos(
    resolution=50,
    lonaxis={"range": [-180, 180]},
    lataxis={"range": [67, -4]}  # Adjusted latitude range
)

fig.update_layout(
    height=800,  # Adjust the height value as needed
    legend=dict(
        orientation="h",  # Set legend orientation to horizontal
        yanchor="bottom",
        y=1.02,
        xanchor="right",
        x=1
    )
)

fig.show()


In [ ]:
import plotly.express as px

fig = px.scatter_geo(
    scope="world",
    data_frame=df1,
    lon=df1.longitude,
    lat=df1.latitude,
    hover_data=df1,
    color=df1.state_country,
    basemap_visible=True,
    color_discrete_sequence=px.colors.qualitative.Set1  # Use a qualitative color scale
)

fig.update_geos(
    resolution=50,
    lonaxis={"range": [-180, 180]},
    lataxis={"range": [67, -4]}
)

fig.update_layout(
    height=800,
    legend=dict(
        orientation="h",
        yanchor="bottom",
        y=1.02,
        xanchor="right",
        x=1
    )
)

fig.show()


In [ ]:
from geopy.geocoders import Nominatim

# Create a Nominatim geolocator
geolocator = Nominatim(user_agent="reverse_geocoding_examples")

# Example coordinates (replace with your actual coordinates)
latitude = -6.379500
longitude =    130.112400

# Perform reverse geocoding
location = geolocator.reverse((latitude, longitude), language='en')

if location is not None:
    address = location.address
    print(f"The location at ({latitude}, {longitude}) is: {address}")
else:
    print(f"No location found for coordinates ({latitude}, {longitude})")


In [ ]:
pip install shiny

In [1]:
conda install ipywidgets=7.7.3


Note: you may need to restart the kernel to use updated packages.
Solving environment: ...working... failed with initial frozen solve. Retrying with flexible solve.

Solving environment: ...working... failed with initial frozen solve. Retrying with flexible solve.


PackagesNotFoundError: The following packages are not available from current channels:

  - ipywidgets=7.7.3

Current channels:

  - https://repo.anaconda.com/pkgs/main/win-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/win-64
  - https://repo.anaconda.com/pkgs/r/noarch
  - https://repo.anaconda.com/pkgs/msys2/win-64
  - https://repo.anaconda.com/pkgs/msys2/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.




In [3]:
conda install -c conda-forge ipywidgets=7.7.3

^C

Note: you may need to restart the kernel to use updated packages.


In [5]:
pip install ipywidgets==7.7.3

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.



Note: you may need to restart the kernel to use updated packages.



CommandNotFoundError: No command 'conda ipywidgets'.

